python/hail/linalg/rowmatrix.py

```diff
@@ -0,0 +1,251 @@
+from hail.utils.java import Env, joption
+from hail.typecheck import *
+from hail.expr.expressions import expr_float64, matrix_table_source, check_entry_indexed
+
+row_matrix_type = lazy()
+
+
+class RowMatrix(object):
+    """Hail's row-distributed matrix of :py:data:`.tfloat64` elements.
+
+    .. include:: ../_templates/experimental.rst
+
+    Notes
+    -----
+    Row matrices must have fewer than :math:`2^31` columns.
+
+    Under the hood, row matrices are partitioned into group of rows.
+    """
+
+    def __init__(self, jrm):
+        self._jrm = jrm
+
+    @property
+    def n_rows(self):
+        """Number of rows.
+
+        Returns
+        -------
+        :obj:`int`
+        """
+        return self._jrm.nRows()
```

Comment:
If computing the number of rows is expensive, I think you should cache the result as a private property of the class.

Reply:
The Scala class uses caching.